libsolidity/ast/ASTJsonConverter.cpp

```
@@ -270,6 +270,7 @@ bool ASTJsonConverter::visit(ContractDefinition const& _node)
 		make_pair("abstract", _node.abstract()),
 		make_pair("baseContracts", toJson(_node.baseContracts())),
 		make_pair("contractDependencies", getContainerIds(_node.annotation().contractDependencies, true)),
+		make_pair("emittedEvents", getContainerIds(_node.interfaceEvents(true))),
```

Why does this belong to the AST btw?

---

It is important information like the `contractDependencies` above. The AST with its annotations contains much more than jus the syntactical information.

---

Why do we ignore errors here?

---

I think it might be because the call graph annotations are not available if you stop after parsing and don't run the analysis.

---

Oh, the reason is actually different: https://github.com/ethereum/solidity/pull/10996#issuecomment-786026787